server/manageuser.php

```diff
@@ -0,0 +1,178 @@
+<?php
+
+require_once 'config.php';
+require_once "user.class.php";
+
+$USER = new User();
+
+if (!$USER->hasPermission('edit_user_permission')) {
+	header("Location: /unauthorized");
+	die();
+}
+
+if(isset($_REQUEST['callback'])){
+	switch ($_REQUEST['callback']) {
+		case 'getUserTable':
+			getUserTable($_REQUEST);
+			break;
+		case 'updateUserPermissions':
+			updateUserPermissions($_REQUEST);
+			break;
+		case 'addUser':
+			addUser($_REQUEST);
+			break;
+		default:
+			die('Unregistered command. This instance has been reported.');
+			break;
+	}
+}
+
+function getUserTable($data){
+	GLOBAL $DB_CONN;
+
+	$response = array();
+	$response['success'] = true;
+
+	$stmt = $DB_CONN->prepare("SELECT userId, firstName, lastName, email, preparesTaxes, archived 
+		FROM vita.user
+		ORDER BY firstName, lastName");
+
+	$stmt->execute(array());
+
+	$thead = "<thead><tr><th>Name</th><th>Email</th><th>Permissions</th><th>Count</th></tr>";
+	$tbody = "<tbody>";
+	while($row = $stmt->fetch(PDO::FETCH_ASSOC)){
+		$optionList = getUserPermissionOptionList($row['userId']);
+
+		$tbody.= "<tr data-user-id='".$row['userId']."'>";
+		$tbody.= "<td>".$row['firstName']." ".$row['lastName']."</td>";
+		$tbody.= "<td>".$row['email']."</td>";
+		$tbody.= "<td><select class='userPermissionList selectpicker' multiple=true>".implode('', $optionList['options'])."</select></td>";
```

Comment:
I really don't like how you're specifying these CSS classes in this PHP code.. Do you think it would be better to just return the data back to the AJAX request, and then in the success function, create the table and append everything? Or why did you take this approach?

Reply:
I've done both(building HTML in JS and building HTML in PHP) enough to where I prefer PHP because it's more convenient. Template literals in JS are pretty nice if forced to build HTML client side. I build it inline in the HTML nowadays using VueJS, which I prefer even more. 
I can do client side if needed here.

Reply:
Let's not worry about doing it client-side for now. But my argument for client-side would be that we keep the front-end stuff more on the front-end rather than having front-end stuff in what is essentially our back-end. I would prefer more strict lines between layers, than letting them cross like this is doing. If one ever renamed the class, one would need to know that this is happening in the backend PHP which is kinda poor design in my opinion. 

But since this works, let's leave it for now and focus on more important things